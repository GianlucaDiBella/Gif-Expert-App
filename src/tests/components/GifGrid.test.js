import React from 'react';
const { GifGrid } = require("../../components/GifGrid")
const { shallow } = require("enzyme")
import '@testing-library/jest-dom'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente <GifGrid />', () => {
    const category = 'Bob Esponja'
    
    const gifs = [{
        id: 'ABC',
        url: 'https://localhost/imagen.png',
        title: 'Titulo'
    },
    {
        id: 'CBA',
        url: 'https://localhost/imagen.png',
        title: 'Titulo'
    }
]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false
    });

    let wrapper = shallow(<GifGrid category={category}/>)

    beforeEach ( () => {
        let wrapper = shallow(<GifGrid category={category}/>)
    })

   test('El componente debe mostrarse correctamente', () => { 
    //    const wrapper = shallow(<GifGrid category={category}/>)
       expect(wrapper).toMatchSnapshot()
   })

   test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {

        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length)
   })
   
    
})
