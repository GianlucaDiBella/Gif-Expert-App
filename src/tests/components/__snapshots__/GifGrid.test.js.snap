// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> El componente debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
     
    Bob Esponja
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo"
      url="https://localhost/imagen.png"
    />
    <GifGridItem
      id="CBA"
      key="CBA"
      title="Titulo"
      url="https://localhost/imagen.png"
    />
  </div>
</Fragment>
`;
